ec/internal/planmodifiers/use_state_unless_template_changed.go

```diff
@@ -63,6 +64,16 @@ func (m useStateForUnknownUnlessMigrationIsRequired) PlanModifyInt64(ctx context
 func (m useStateForUnknownUnlessMigrationIsRequired) UseState(ctx context.Context, configValue attr.Value, plan tfsdk.Plan, state tfsdk.State, planValue attr.Value, stateValue attr.Value) (bool, diag.Diagnostics) {
 	var diags diag.Diagnostics
 
+	var parentResState attr.Value
+	diag := state.GetAttribute(ctx, path.Root(m.resourceKind), &parentResState)
+	diags.Append(diag...)
```

Comment:
We need to either check `diag` for an error and return it if it contains one or return it instead of `nil` in all subsequent `return` statements. (the method uses the second way so it accumulates all encountered errors in `diags`).
As a side note, name `diag` hides the corresponding imported package so IMO it's better to rename the error var or better do something like `diags.Append(state.GetAttribute(ctx, path.Root(m.resourceKind), &parentResState)...)`

Reply:
thank you for your comment, indeed that would cause an issue, in this case it would be unexpected that this would generate an error so is tempting to check and return, but again since this is unlikely to happen I think I'll follow the same pattern I'll add the diagnostics to the `diags` variable